Using official WordPress images and introducing new GitHub action [MAILPOET-6097]

lysyjan opened the pull request:
## Description

The goal of this PR is to start using official WordPress docker images in our integration and acceptance tests. This step allows us to create automated updating docker images, which is possible via a new GitHub action. I want to test it manually in the first phase, so I need to merge those changes to the trunk branch.

## Code review notes

- I want to add more comments and polish the scripts in the next pull request when I am sure that the scripts work as I expect.
- In the first phase the action should only commit changes to the specific branch if there are some
- The next pull request will add creating a pull request

## QA notes

We can skip QA here.

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-6097]

## After-merge notes

_N/A_

## Tasks

- [x] 🚫 I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [x] 🚫 I added sufficient test coverage
- [x] 🚫 I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes


[MAILPOET-6097]: https://mailpoet.atlassian.net/browse/MAILPOET-6097?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

lysyjan committed:
Use official WP docker images in acceptance and integration tests

[MAILPOET-6097]

lysyjan committed:
Add new GitHub action for using newer WP image

[MAILPOET-6097]

lysyjan committed:
Add new GitHub action updating used WooCommerce plugin

[MAILPOET-6097]

lysyjan committed:
Create file with helper functions

To avoid code repetition I created a file,
which contains function that can be reused.
[MAILPOET-6097]

lysyjan committed:
Add script updating used WC Memberships

[MAILPOET-6097]

lysyjan committed:
Add script updating used Automate Woo

[MAILPOET-6097]

lysyjan committed:
Add script updating used WC Subscriptions

[MAILPOET-6097]

lysyjan committed:
Use all updating scripts

We need to use all scripts and also use different IDs.
Some scripts also need to use token stored in GH secrets.
[MAILPOET-6097]

lysyjan committed:
Update used latest WooCommerce Memberships plugin

We were using older Memberships plugin version which caused throwing deprecated errors.
[MAILPOET-6097]

costasovo reviewed (commented):
@lysyjan Looks good! Thank you for adding these automations!

You mentioned you are going to test it on trunk, so I didn't test anything this time. I don't even know how :)

I think that this commit https://github.com/mailpoet/mailpoet/pull/5659/commits/c911859cce0a5f0f255eb061da49850acb6ef6fd has an incorrect message. It says: "Add new GitHub action updating used WordPress images" but it is about updating WooCommerce plugin. Could you please update the message?

lysyjan commented:
> I think that this commit https://github.com/mailpoet/mailpoet/commit/c911859cce0a5f0f255eb061da49850acb6ef6fd has an incorrect message. It says: "Add new GitHub action updating used WordPress images" but it is about updating WooCommerce plugin. Could you please update the message?

You were right. I have updated it.

> You mentioned you are going to test it on trunk, so I didn't test anything this time. I don't even know how :)

From what I know, it is not possible to execute action if it is not in the trunk branch.

lysyjan committed:
Use already configured token

We have already set existing GH token in the repository settings, so we can reuse it as well.
[MAILPOET-6097]
